frontend/src/state/index.ts

```diff
@@ -187,6 +192,31 @@ export const getCourseWarningsFromState = (
   );
 };
 
+/**
+ * Gets the name of the course with the dndId in a certain semester in the active plan
+ * @param state the current state
+ * @param dndId the id of the course to search for
+ * @param semester the semester the course exists in
+ * @returns the name of that course or empty string if can't find
+ */
+export const getCourseNameFromState = (
+  state: AppState,
+  dndId?: string,
+  semester?: ScheduleTerm
```

Comment:
We made these option because we had to call `getCourseNameFromState` from ClassBlockBody.tsx, and dndId and semester are optional params